rio_cogeo/writer.py

```diff
@@ -0,0 +1,44 @@
+import os
+import boto3
+from rasterio.env import getenv
+from rasterio.shutil import copy
+
+
+SCHEMES = {
+    's3': {
+        's3:': 2,
```

Comment:
Shouldn't the first value be 1?
```suggestion
        's3:': 1,
```
Could possibly implement this dict with an Enum too.

Reply:
No, I use this dict to parse the `dst_path`. So I split it by `/` delimiter and take according to this the bucket and the file name. So the paths could be:
- `s3://bucket-name/file-name` - so the bucket name is [2]
- `/vsis3/bucket-name/file-name` - [2]
- `https://s3.amazonaws.com/bucket-name/file-name` - [3]